C/include/c4ReplicatorTypes.h

```diff
@@ -203,6 +203,7 @@ typedef struct C4ReplicatorParameters {
     const C4SocketFactory* C4NULLABLE                 socketFactory;    ///< Custom C4SocketFactory, if not NULL
     C4ReplicationCollection*                          collections;
     size_t                                            collectionCount;
+    const C4KeyPair* C4NULLABLE                       externalKey;
```

Comment:
It doesn't make sense to use `const C4KeyPair*`, since C4KeyPair is immutable. (There are a bunch of other places this appears later on, too.)

Reply:
Will remove the const.
As for Retained, the compiler wants to see the complete definition, instead of forward declaration. The entire class definition of C4KeyPair is under #ifdef COUCHBASE_ENTERPRISE

Reply:
One thing I have noticed in `C4Cert` is that the base declaration of the `C4Cert` struct, including its inheritance is outside `COUCHBASE_ENTERPRISE`. If doing the same for C4KeyPair, will it solve the problem here?

```
struct C4Cert final
    : public fleece::RefCounted
    , public fleece::InstanceCountedIn<C4Cert>
    , C4Base {
#ifdef COUCHBASE_ENTERPRISE
...
#endif  // COUCHBASE_ENTERPRISE
```

Reply:
This will suddenly expose a class definition of C4KeyPair to the CE. Currently, the only thing CE sees about C4KeyPair is a opaque declaration, except for c4keypair_retain ad c4keypair_release. (Don't whether it's intentional.) I think the cleaner way could be putting "#ifdef" around "externalKey" every to please the compiler. Although there is no way C4KeyPair can be created in CE, but the compiler won't compile if any functions of C4KeyPair is used except for _retain and _release.

Reply:
Basically, the code we are concerned about, using unique_ptr in lieu of Retained would not be hit in CE. We may put ENTERPRISE in C4ReplicatorParameters, or start the "#ifdef" in Core/c4RemoteReplicator and downward just to get compiled.

Reply:
> This will suddenly expose a class definition of C4KeyPair to the CE

I think this is fine as we already did it for C4Cert?. Also as you mentioned, c4keypair_retain / release is also visible to CE already. If feels weird when having C4KeyPair passing through `C4ReplicatorParameters` when C4KeyPair definition is not actually available. I cannot comment about #ifdef `externalKey`.

> We may put ENTERPRISE in C4ReplicatorParameters, or start the "#ifdef" in Core/c4RemoteReplicator and downward just to get compiled.

This is counterpart alternative.